src/modules/directory.rs

```diff
@@ -15,28 +14,25 @@ use std::path::Path;
 ///
 /// **Truncation**
 /// Paths will be limited in length to `3` path components by default.
-pub fn segment(_: &ArgMatches) -> Option<Segment> {
+pub fn segment(current_dir: &Path, _args: &ArgMatches) -> Option<Segment> {
     const HOME_SYMBOL: &str = "~";
     const DIR_TRUNCATION_LENGTH: usize = 3;
     const SECTION_COLOR: Color = Color::Cyan;
 
     let mut segment = Segment::new("dir");
 
-    // TODO: Currently gets the physical directory. Get the logical directory.
-    let current_path = env::current_dir().expect("Unable to identify current directory");
-
     let dir_string;
-    if let Ok(repo) = git2::Repository::discover(&current_path) {
+    if let Ok(repo) = git2::Repository::discover(current_dir) {
         // Contract the path to the git repo root
         let repo_root = get_repo_root(&repo);
         let repo_folder_name = repo_root.file_name().unwrap().to_str().unwrap();
 
-        dir_string = contract_path(&current_path, repo_root, repo_folder_name);
+        dir_string = contract_path(current_dir, repo_root, repo_folder_name);
```

Comment:
What do the &variables do and why did you remove these?

Reply:
`&variables` are references to an object, rather than objects themselves.
It's good practice to use instances where possible as it's lighter in memory to pass a reference around than an instance.

https://github.com/starship/starship/pull/6/files/daeb7cf6bfbd170f149ea98a73bfa5d5a5b67a07#diff-95036686adb679a23cd72cd4cca8cbc9R17
In this case (`current_dir: &Path`) `current_dir` is a reference to a `Path`, so I don't need to add the ampersand. If instead, the function param accepted `current_dir: Path`, I would refer to it with `&current_dir` so that I was using the reference.

Is that clear? 🙂